### PR TITLE
Introduce basic PMD rules

### PR DESCRIPTION
**Pull request by emanuelez**
* Moves config files up one folder (so they can be used by the annotation processor in the future)
 * Introduces PMD basic PMD checks and fixes the issues found

### CODE REVIEW

**Comment by emanuelez** [comment]
@realm/java

File: realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java [hunk 1]
```diff
@@ -222,4 +222,9 @@ public boolean equals(Object o) {
         return o instanceof RealmObservableFactory;
     }
 
+    @Override
+    public int hashCode() {
+        return 37;
```
Comment (kneth) [inline realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java]: Really?

**Comment by kneth** [comment]
Nice catches!

File: realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java (hunk 1 above)
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java]: The class has no state that can be set from the outside, so any instance of this class is equal to any other instance of this class. A hardcoded hashCode is fine in that case :)

**Comment by emanuelez** [comment]
retest this please

**Comment by emanuelez** [comment]
retest this please

**Comment by emanuelez** [comment]
retest this please

**Comment by cmelchior** [comment]
:+1:

**Pull request merged**